src/sagemaker_containers/serializers.py

```diff
@@ -0,0 +1,185 @@
+
+        Args:
+            nparray (np.array): np array object to be serialized.
+
+        Returns:
+            (obj): serialized np array.
+        """
+        buffer = BytesIO()
+        np.save(buffer, nparray)
+        return buffer.getvalue()
+
+
+class Json(object):
+    """JSON serializer"""
+
+    @staticmethod
+    def loads(string):  # type: (str) -> object
+        """Deserialize a JSON string in a Python object.
+
+        Args:
+            string (str): JSON string to be deserialized
+
+        Returns:
+            (object): deserialized Python object.
+        """
+        return stdjson.loads(string)
+
+    @staticmethod
+    def _numpy_dumps(obj):
+        try:
+            return stdjson.dumps(obj.tolist())
+        except (TypeError, SyntaxError, AttributeError):
+            return None
+
+    @staticmethod
+    def _stream_dumps(obj):
+        try:
+            return stdjson.dumps(obj.read())
+        except (TypeError, SyntaxError, AttributeError):
+            return None
+
+    def dumps(self, obj):  # type: (object) -> str
+        """Serializes python objects, streams, numpy arrays into JSON.
+
+        Args:
+            obj (object): object to be serialized.
+
+        Returns:
+            str:  serialized object
+        """
+        return self._numpy_dumps(obj) or self._stream_dumps(obj) or stdjson.dumps(obj)
+
+
+class Csv(object):
+    """JSON serializer"""
+
+    @staticmethod
+    def loads(string):  # type: (str) -> object
+        """Deserialize a CSV string in a Python object.
+
+        Args:
+            string (str): CSV string to be deserialized
+
+        Returns:
+            (object): deserialized Python object.
+        """
+        try:
+            string = string.decode('utf-8')
+        except AttributeError:
+            pass
+        stream = StringIO(string)
+        return np.genfromtxt(stream, dtype=np.float32, delimiter=',')
+
+    @staticmethod
+    def dumps(obj):  # type: (object) -> str
+        """Serializes python objects, streams, numpy arrays into CSV.
+
+        Args:
+            obj (object): object to be serialized.
+
+        Returns:
+            str:  deserialized object.
+        """
+        stream = StringIO()
+        np.savetxt(stream, obj, delimiter=',', fmt='%s')
+        return stream.getvalue()
+
+
+def dumps(obj, content_type):
+    """Serializes an object to a one of the default content types.
+
+    Args:
+        obj (object): to be serialized.
+        content_type (str): content type to be used.
+
+    Returns:
+        object: serialized object.
+    """
+    return _SerializerMap(content_type).dumps(obj)
+
+
+def loads(obj, content_type):
+    """Deserialize an object to a one of the default content types.
+
+    Args:
+        obj (object): to be deserialized.
+        content_type (str): content type to be used.
+
+    Returns:
+        object: deserialized object.
+    """
+    return _SerializerMap(content_type).loads(obj)
+
+
+npy = Npy()
```

Comment:
module level... why?

Reply:
Yeah, I am not sold on that one. Trying to think a cleaner solution.

Reply:
eliminate the declarations. create the instances inside _SerializerMap.__init__ (but see comment on that class below)